tests/unit/select_test.coffee

```diff
@@ -344,3 +344,22 @@ test 'optionValuePath with nested valuePath', ->
 
   equal obj2, select.get('selection'),
     'The right selection is retrieved'
+
+test 'shouldEnsureVisible controls whether to ensure visibility', ->
+  expect 2
+
+  select = @subject
+    content: ['foo']
+    ensureVisible: ->
+      ok yes, 'ensureVisible is called if shouldEnsureVisible is true'
+  select.set 'highlighted', 'foo'
+
+  select = @subject
+    content: ['bar']
+    shouldEnsureVisible: no
+
+  spy = sinon.spy select, 'ensureVisible'
+  select.set 'highlighted', 'bar'
+  equal spy.callCount, 0,
+    'ensureVisible is not called if shouldEnsureVisible is false'
+  spy.restore()
```

Comment:
add `spy = null` outside of the test at top of the file (like the `colorPicker` here: https://github.com/Addepar/ember-widgets/blob/master/tests/unit/color_picker_test.coffee)

Then in `teardown` check if spy is not null restore it.